react/components/ExtensionPoint/ComponentLoader.tsx

```diff
@@ -60,16 +68,64 @@ const ComponentLoader: FunctionComponent<Props> = props => {
     hydration,
   } = props
 
+  /**
+   * Slot props should ALWAYS be PascalCased.
+   * It is OK to not include componentProps in the dependency array
+   * since there is currently no way for users to ADD or UPDATE slots via CMS.
+   * What this means is that the slots variable below only needs to be
+   * computed once during runtime, since we know that, even if componentProps
+   * is updated, the props that function as Slots will NOT change.
+   */
+  const slots = useMemo(() => {
+    if (!componentProps) {
+      return {}
+    }
+
+    const slotNames = Object.keys(componentProps).filter(
+      key => key[0] !== key[0].toLowerCase()
+    )
+    const resultingSlotsProps: Record<string, ReactNode> = {}
+
+    for (const slotName of slotNames) {
+      resultingSlotsProps[slotName] = generateSlot({
+        treePath,
+        slotName,
+        slotValue: componentProps[slotName],
+        hydration,
+      })
+    }
+
+    return resultingSlotsProps
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [hydration, treePath])
+
+  const componentPropsWithSlots = useMemo(
+    () => ({
+      ...componentProps,
+      ...slots,
+    }),
+    [componentProps, slots]
+  )
+  const asyncComponentProps = useMemo(
+    () => ({
+      ...props,
+      props: componentPropsWithSlots,
+    }),
+    [componentPropsWithSlots, props]
+  )
+
+  const Component = useMemo(() => component && getImplementation(component), [
+    component,
+  ])
```

Comment:
`getImplementation` is just an access to an object this useMemo is kind of useless, or am I missing something?

```suggestion
  const Component = component && getImplementation(component)
```

Reply:
Yep, you're right! I added this while trying to debug the unnecessary re-renders on this component, but I guess it is not really necessary :)